.github/workflows/ci.yml

```diff
@@ -0,0 +1,47 @@
+name: CI (Build)
+
+on:
+  # Triggers the workflow on push or pull request events
+  push:
+    branches:
+      - "master"
+  pull_request:
+    branches:
+      - "*"
+  # Allows running this workflow manually from the Actions tab
+  workflow_dispatch:
+
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```

Comment:
@jcfr Do you think worthwhile running build tests on the other platforms (macOS and Windows) as well? I believe GitHub hosted runners are free for public repositories.

Reply:
Building and testing(*) CTKCore/CTKWidget along with associated wrapping across platform should be straightforward.

It becomes more challenging to test quickly against DCMTK & VTK, I think we could have a scheduled workflow running the CTK superbuild once a week but having it for each pull request would be time consuming.

There are also cross-platform solution for installing dependencies (e.g vcpkg) we could look into.

The time being, I will aim for having something reasonable.

(*) testing on linux will likely require some trick.